main/aro-provision-install-test-pipeline.yaml

```diff
@@ -0,0 +1,325 @@
+          value: $(params.gateway-crd)
+        - name: keycloak-channel
+          value: $(params.keycloak-channel)
+        - name: helm-install-flags
+          value: $(params.helm-install-flags)
+      runAfter:
+        - helm-uninstall
+        - prepare-for-rhcl-rc-install
+      taskRef:
+        kind: Task
+        name: helm-install
+      workspaces:
+        - name: shared-workspace
+          workspace: shared-workspace
+    - name: run-tests
+      params:
+        - name: testsuite-image
+          value: $(params.testsuite-image)
+        - name: project
+          value: $(params.project)
+        - name: make-target
+          value: $(params.make-target)
+        - name: pytest-flags
+          value: $(params.pytest-flags)
+        - name: settings-cm
+          value: $(params.settings-cm)
+        - name: additional-env
+          value: $(params.additional-env)
+        - name: kubeconfig-path
+          value: $(tasks.kubectl-login.results.kubeconfig-path)
+        - name: cluster-credentials
+          value: $(tasks.provision-aro.results.credentials-secret)
+      runAfter:
+        - helm-install
+      taskRef:
+        kind: Task
+        name: run-tests
+      workspaces:
+        - name: shared-workspace
+    - name: upload-results
+      when:
+        - input: $(params.upload-results)
+          operator: in
+          values: ["true"]
+      params:
+        - name: launch-name
+          value: $(params.launch-name)
+        - name: launch-description
+          value: $(params.launch-description)
+        - name: testsuite-image
+          value: $(params.testsuite-image)
+        - name: make-target
+          value: $(params.make-target)
+        - name: rp-project
+          value: $(params.rp-project)
+        - name: ocp-version
+          value: $(tasks.kubectl-login.results.ocp-version)
+      runAfter:
+        - run-tests
+      taskRef:
+        kind: Task
+        name: upload-results
+      workspaces:
+        - name: shared-workspace
+    - name: helm-uninstall-cleanup
+      when:
+        - input: $(params.cleanup)
+          operator: in
+          values: ["true"]
+      params:
+        - name: kubeconfig-path
+          value: $(tasks.kubectl-login.results.kubeconfig-path)
+      runAfter:
+        - upload-results
+        - run-tests
+      taskRef:
+        kind: Task
+        name: helm-uninstall
+      workspaces:
+        - name: shared-workspace
+          workspace: shared-workspace
+    - name: delete-aro
+      when:
+        - input: $(params.cleanup)
+          operator: in
+          values: ["true"]
+      params:
+        - name: cluster-name
+          value: $(params.cluster-name)
+        - name: azure-credentials
+          value: $(params.azure-credentials)
+      runAfter:
+        - helm-uninstall-cleanup
+      taskRef:
+        kind: Task
+        name: delete-aro
+      workspaces:
+        - name: shared-workspace
+    - name: cluster-secret-cleanup
```

Comment:
This task doesn't use workspace with saved kubeconfig and removes secret from already removed cluster? How does that work here?

Reply:
This pipeline runs on some cluster (let's call it Hub Cluster). It provisions ARO cluster and creates a secret with credentials to this (newly provisioned) ARO cluster in Hub Cluster. Once ARO cluster is removed this credentials secret in Hub Cluster is of no use anymore and it also needs to be removed as part of cleanup.

TaskRuns have access to cluster they are running on (Hub Cluster) automatically so no need to setup any kubeconfig.

Reply:
Makes sense, thanks for the explanation